### DISCUSSION TITLE
Cursorword but for instances instead of just words

### DISCUSSION THREAD

**Discussion by carlesba** [nvim-mini/mini.nvim#931] [Ideas]
I would find more useful if cursorword module highlights instances of the current variable in the cursor instead of just copies of this word. It should be achievable via treesitter.

Additionally, a command to jump between those instances would be huge.

**Comment by echasnovski** [comment]
Thanks for the suggestion!

'mini.cursorword' will keep highlighting only based on the text (word under cursor, a.k.a. `<cword>`).

What you are asking for is not really achievable with tree-sitter (as it does not really keep connections between nodes, only their structure). It *is* doable with LSP., though. On Neovim 0.10 there is a `vim.lsp.buf.document_highlight()` function which highlights based on the current position with the help of LSP server. See `:h vim.lsp.buf.document_highlight()` for the basic setup it can provide.

**Reply by carlesba** [reply]
Great. Thanks for the tip!
